pkg/cli/logfile/logfile.go

```diff
@@ -81,16 +89,25 @@ func Hook() error {
 	cmd.Env = os.Environ()
 	cmd.Env = append(cmd.Env, fmt.Sprintf("%s=1", EnvironmentVariable))
 
+	l, err := net.Listen(TraceSocketType, "localhost:0")
```

Comment:
Fold the `net.Listen` and `newTraceServer` into `newRecorder`? This would allow us to duplicate less code as well as make it easier to shut everything down / wait for it to finish

Reply:
I kept the listener creation here because we need it to set the environment variable and it does not result in any code duplication, but I moved all the management to the recorder.

Reply:
Let me take a quick pass at it and see if it feels better, I'll let this land since it's going into my logfile PR anyways tho :D